frontend/packages/ux-editor/src/components/config/expressions/DataSourceValue.tsx

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import { useParams } from 'react-router-dom';
+import { useSelector } from 'react-redux';
+import { Select, TextField, ToggleButtonGroup } from '@digdir/design-system-react';
+import { DataSource } from '../../../types/Expressions';
+import { ExpressionElement } from "./ExpressionContent";
+import { IFormLayouts } from '../../../types/global';
+import { DatamodelFieldElement } from 'app-shared/types/DatamodelFieldElement';
+import { FormComponent } from '../../../types/FormComponent';
+import { useDatamodelMetadataQuery } from '../../../hooks/queries/useDatamodelMetadataQuery';
+import { useFormLayoutsQuery } from '../../../hooks/queries/useFormLayoutsQuery';
+import { selectedLayoutSetSelector } from '../../../selectors/formLayoutSelectors';
+
+interface DataSourceValueProps {
+  expressionElement: ExpressionElement;
+  currentDataSource: DataSource;
+  specifyDataSourceValue: (dataSourceValue: string) => void;
+  isComparableValue: boolean;
+  onSetDuplicatedComponentIdsDiscovered: (value: boolean) => void;
+}
+
+export const DataSourceValue = ({
+                                  expressionElement,
+                                  currentDataSource,
+                                  specifyDataSourceValue,
+                                  isComparableValue,
+                                  onSetDuplicatedComponentIdsDiscovered
+                                }: DataSourceValueProps) => {
+  const { org, app } = useParams();
+  const selectedLayoutSet = useSelector(selectedLayoutSetSelector);
+  const datamodelQuery = useDatamodelMetadataQuery(org, app);
+  const formLayoutsQuery = useFormLayoutsQuery(org, app, selectedLayoutSet);
+  const dataModelElementsData = datamodelQuery?.data ?? [];
+  const formLayoutsData = formLayoutsQuery?.data ?? [];
+
+  const getDataModelElementNames = (dataModelElements: DatamodelFieldElement[]) => {
+    return dataModelElements
+      .filter(element => element.dataBindingName)
+      .map((element) => ({
+        value: element.dataBindingName,
+        label: element.dataBindingName,
+      }))
+  };
+
+  const findDuplicatedIds = (arr) => {
+    const idOccurrences = arr.reduce((occurrences, compId) => {
+      occurrences[compId] = (occurrences[compId] || 0) + 1;
+      return occurrences;
+    }, {});
+
+    return Object.keys(idOccurrences).filter(id => idOccurrences[id] > 1);
+  };
+
+  const getUniqueComponentIds = (formLayouts: IFormLayouts) => {
+    const components = Object.values(formLayouts).flatMap(layout => Object.values(layout.components));
```

Comment:
Am I missing the componentIDs within groups by doing this?

Reply:
not sure , but maybe we need to go through layout and check first if we have components and group